core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -53,19 +52,9 @@ class Project(
   }
   private val classfileWatcher = context.actorOf(Props(new ClassfileWatcher(config, searchService :: reTypecheck :: Nil)), "classFileWatcher")
 
-  def receive: Receive = awaitingConnectionInfoReq
+  def receive: Receive = handleRequests
 
-  def awaitingConnectionInfoReq: Receive = withLabel("awaitingConnectionInfoReq") {
-    case ConnectionInfoReq =>
-      sender() ! ConnectionInfo()
```

Comment:
You misunderstand the change in logic - its handled in the handleRequests method
The original code was a hack to say 'handle this one, and trigger indexing, stash anything until we have seen it.